src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -51,109 +49,42 @@ public CommandResult execute(Model model) throws CommandException {
+                }
+                Person person = CsvPersonParser.parsePerson(cleanedFields, model);
                 newPersons.add(person);
             }
-        } catch (IOException | CsvValidationException e) {
-            throw new CommandException("Error reading from the CSV file: " + e.getMessage());
-        }
-        model.replaceAllPersons(newPersons);
-
-        return new CommandResult(String.format("Successfully imported %d persons.", newPersons.size()));
-    }
 
-
-    /**
-     * Parses a person from CSV fields.
-     */
-    private Person parsePerson(String[] fields, Model model) throws CommandException {
-        try {
-            Name name = ParserUtil.parseName(fields[0].trim());
-            Phone phone = ParserUtil.parsePhone(fields[1].trim());
-            Email email = ParserUtil.parseEmail(fields[2].trim());
-            Telegram telegram = ParserUtil.parseTelegram(fields[3].trim());
-            Github github = ParserUtil.parseGithub(fields[5].trim());
-
-            // Process tags
-            Set<Tag> tags = parseTags(fields[4].trim());
-
-            Map<String, Assignment> assignment = parseAssignment(fields[6].trim(), model);
-
-            Set<Integer> weeksPresent = new HashSet<>();
-            if (!fields[7].trim().isEmpty()) {
-                String[] weeksArray = fields[7].split(",");
-                for (String week : weeksArray) {
-                    int weekPresent = ParserUtil.parseWeek(week.trim());
-                    weeksPresent.add(weekPresent);
-                }
-            }
-
-            return new Person(name, phone, email, telegram, github, assignment, weeksPresent, tags);
-
-        } catch (IllegalArgumentException | ParseException e) {
-            throw new CommandException(e.getMessage());
-        }
-    }
-
-    /**
-     * Parses a set of tags from a string with tags in the format "[tag1],[tag2]".
-     */
-    private Set<Tag> parseTags(String tagField) {
-        Set<Tag> tags = new HashSet<>();
-
-        // Remove surrounding brackets for each tag
-        tagField = tagField.trim();
-        if (!tagField.isEmpty()) {
-            // Split on commas first, then remove the square brackets from each tag
-            String[] tagArray = tagField.split(",");
-            for (String tag : tagArray) {
-                tag = tag.trim(); // Remove any extra spaces
-                if (tag.startsWith("[") && tag.endsWith("]")) {
-                    tag = tag.substring(1, tag.length() - 1); // Remove the brackets
-                }
-                if (!tag.isEmpty()) {
-                    tags.add(new Tag(tag));
-                }
+            if (newPersons.isEmpty()) {
+                throw new CommandException("There is no person data present");
             }
+            model.replaceAllPersons(newPersons);
+        } catch (IOException | CsvValidationException | DuplicatePersonException | CommandException e) {
```

Comment:
I think this can be better if we separate the exceptions out and handle them separately.